src/odemis/gui/comp/grid.py

```diff
@@ -127,19 +127,25 @@ def get_2_viewports(self):
         if [view.Shown for view in self.viewports].count(True) > 2:
             logging.warning("Found more than two viewports to show, only the first 2 are displayed in this layout.")
 
-        for v in self.viewports:
-            try:
-                if v.view is not None and v.Shown:
+        if len(self.visible_viewports) == 2:
+            logging.debug("Only two viewports are visible so these are shown.")
+            return self.visible_viewports
+
+        else:
+            # If more/less than 2 viewports are visible use the first 2 viewports with a view
+            for v in self.viewports:
+                try:
+                    if v.view is not None and v.Shown:
+                        viewports.append(v)
+                except AttributeError:
+                    # Should never happen, unless it's not really a ViewPort.
+                    # If so, let's not go completely bad
+                    logging.exception("Viewport has no view")
```

Comment:
Can you add a `"%s", viewport` as info to the log message so that it's a little bit clearer what happened in such case.
Ideally, could you also update the log messages in the other cases. (for instance line 114, which is even more confusing for now)